sqlitecluster/SQLiteClusterMessenger.cpp

```diff
@@ -232,3 +232,17 @@ bool SQLiteClusterMessenger::runOnLeader(BedrockCommand& command) {
 
     return true;
 }
+
+bool SQLiteClusterMessenger::commandWillCloseSocket(BedrockCommand& command) {
+    // See if either the client or the leader specified `Connection: close`.
+    // Technically, we shouldn't need to care if the client wants to close the connection, we could still re-use the connection from this server to leader, except that we've already sent
+    // it a command with `Connection: close` on this socket so we should expect that it will honor that and close this socket.
+    for (const auto& message : {command.request.nameValueMap, command.response.nameValueMap}) {
+        auto connectionHeader = message.find("Connection");
+        if (connectionHeader != message.end() && connectionHeader->second == "close") {
+            return false;
+        }
+    }
+
+    return true;
```

Comment:
Your logic is inverted here - if we find a `Connection: close` header, the command will close the socket, `commandWillCloseSocket` should return `true`